### PR TITLE
Replace deprecated distutils.version.StrictVersion.

### PR DESCRIPTION
**Pull request by znewman01**
distutils is deprecated as of Python 3.10. We were only using it for `StrictVersion`, which parses a version string and lets you order the result.

The [official recommendation] from the packaging team is to replace `StrictVersion` with a dependency on the `packaging` package, but:

1. The Python version parser was a little different from GPG versioning anyway because it's made for Python versions (it had support for "prerelease tags").
2. I wanted to avoid adding dependencies if possible.

[official recommendation]: https://github.com/pypa/packaging/issues/520

This PR adds a new `Version` class in `securesystemslib.gpg.util` and uses that instead.

It's a breaking change to `gpg.util.get_version()`, but AFAICT that wasn't ever intended for public consumption.

Signed-off-by: Zachary Newman <z@znewman.net>

Fixes: N/A

### Please verify and check that the pull request fulfils the following requirements:

- [X] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [X] Tests have been added for the bug fix or new feature
- [X] Docs have been added for the bug fix or new feature

### CODE REVIEW

**Comment by jku** [comment]
This may be off topic for your PR but since you're touching this you may have an opinion: I wonder how much work would it be to use an actual API for the ~2 gpg commands we need? Spawning new processes via a command line makes me feel dirty (and is a great source of testing trouble).

**Comment by joshuagl** [comment]
Some related discussion in https://github.com/secure-systems-lab/securesystemslib/issues/428 
tl;dr – we don't want to run gpg commands, but we don't want to add dependencies either. Discussion in #428 strives to find a middle ground.

**Comment by znewman01** [comment]
> Spawning new processes via a command line makes me feel dirty (and is a great source of testing trouble).

I'm generally sympathetic to that point of view. While I do think minimizing dependencies is nice, to my mind, there's not much difference between "depends on GPG" and "depends on GPG+Python bindings to GPG."

That said, my most immediate goal is to fix #428 and reduce flakiness in CI for go-tuf; I think the very minor change in #429 would fix it but we need to couple that with some extra testing (#434).

File: tests/test_gpg.py [hunk 1]
```diff
@@ -809,5 +817,30 @@ def test_verify_short_signature(self):
...
+        ('1.3.22', Version(1, 3, 22)),
+    ]:
+      self.assertEqual(Version.from_string(value), expected)
+      self.assertEqual(str(expected), value)
+
+  def test_version_from_string_invalid(self):
+    """Version.from_string rejects invalid inputs."""
+    for value in [
+        '1.3',
+        '1.33.0',
```
Comment (jku) [inline tests/test_gpg.py:837]: These are both valid for StrictVersion.

Obviously we don't need to re-implement StrictVersion here (we know this is a hack, it just has to work well enough for this specific case) but at least the latter one seems strange to me -- is this just because gpg historically hasn't used multidigit minor versions?

**Review by jku — COMMENTED** [review]
Seems reasonable to me, especially if we consider the whole _"library spawns other binaries and parses output"_ as a hack that securesystemslib should probably get rid of.

Left a comment but even that does not seem critical (I don't think it'll break anything in this decade)

**Comment by lukpueh** [comment]
> I wonder how much work would it be to use an actual API like https://pypi.org/project/python-gnupg/ or https://pypi.org/project/gpg/

Too much work for the benefit. See https://github.com/secure-systems-lab/securesystemslib/issues/247 and https://wiki.python.org/moin/GnuPrivacyGuard 

Tl;DR: Both packages you suggest also just spawn a process to run  the `gpg` command, the former via `subprocess`, the latter via GPGME C binding. So far we haven't found a good alternatives to our "hack".

File: tests/test_gpg.py (hunk 1 above)
Comment (lukpueh) [reply tests/test_gpg.py:837]: Good question. Unfortunately, I can't remember if it was just an assumption based on the existing version progression (e.g. last 1.x  was 1.9), or if there was a stronger indication somewhere. Happy to change, although not necessarily in this PR, which just uses the regex that was already there.

**Review by lukpueh — APPROVED** [review]
Great work. Looks more robust, than it was before. Regarding backwards-compatibility, I confirm that `gpg.util.get_version()` was not intended as public API.

File: tests/test_gpg.py (hunk 1 above)
Comment (jku) [reply tests/test_gpg.py:837]: oh you're right: no new bugs then :)
